controllers/controllers.go

```diff
@@ -41,19 +41,59 @@ type instanceReconciler struct {
 	recorder record.EventRecorder
 }
 
-// configureInstance adds the specified instance to the cluster. if hostname is not empty, the instance's hostname will be
-// changed to the passed in value. If annotations is not nil, the node will have the specified annotations applied to
-// it.
-func (r *instanceReconciler) configureInstance(instance *instances.InstanceInfo, annotations map[string]string) error {
+// ensureInstanceIsUpToDate ensures that the given instance is configured as a node and upgraded to the specifications
+// defined by the current version of WMCO. If annotations is not nil, the node will have the specified annotations
+// applied to it.
+func (r *instanceReconciler) ensureInstanceIsUpToDate(instance *instances.InstanceInfo, annotationsToApply map[string]string) error {
+	if instance == nil {
+		return errors.New("instance cannot be nil")
+	}
+
+	// Instance is up to date, do nothing
+	if instance.Node != nil && instanceUpToDate(instance.Node.Annotations) {
+		return nil
+	}
```

Comment:
Re-reading this, I think the function `instanceUpToDate()` should take `InstanceInfo` as the input and you should do all the checks in the function.